analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -370,23 +370,23 @@ private class Graph(private val nodeMap: MutableMap<Identifier, Set<Identifier>>
      * of the given [root] package.
      */
     fun toPackageReferenceForest(root: Identifier): SortedSet<PackageReference> {
-        fun getPackageReference(id: Identifier, predecessorNodes: Set<Identifier> = mutableSetOf()): PackageReference {
-            val currentDependencies = dependencies(id) - predecessorNodes
-            val nextPredecessorNodes = predecessorNodes + currentDependencies
-
-            val dependencyReferences = currentDependencies.mapTo(sortedSetOf()) {
-                getPackageReference(it, nextPredecessorNodes)
-            }
+        fun getPackageReference(id: Identifier, predecessorNodes: Set<Identifier> = emptySet()): PackageReference {
+            val dependencies = nodeMap.getValue(id).mapNotNull {
+                if (predecessorNodes.contains(it)) {
+                    null
+                } else {
+                    getPackageReference(it, predecessorNodes + id)
+                }
```

Comment:
Could be something like

    mapNotNull { ids ->
    // ...
    predecessorNodes.takeUnless { it in ids }?.let { getPackageReference(ids, it + id) }

Reply:
I've made it a separate commit on top, to keep this commit only doing the revert.